src/changelog/linker/mapper/leadingv.go

```diff
@@ -0,0 +1,82 @@
+package mapper
+
+import (
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/Masterminds/semver"
+	"github.com/newrelic/release-toolkit/src/changelog"
+	"github.com/newrelic/release-toolkit/src/changelog/linker"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	checkTimeoutSeconds = 1
+)
+
+// LeadingVCheck wraps another mapper and performs a check over the result of Map. If the check is not
+// successful, it tries to generate it prepending a leading v to the version.
+type LeadingVCheck struct {
+	mapper    linker.Mapper
+	checkLink func(link string) bool
```

Comment:
I think it is dangerous to reduce the signature of a link check to just a single bit of information (`bool`). Namely, the main problem I see with this is that a network error (e.g. github outage) will be indistinguishable from a tag not existing.

I think we could change this to
```suggestion
	checkLink func(link string) (bool, error)
```

Where errors would be network/transport errors, and modify the code so it logs those very verbosely (Errorf), so users can at least notice that a check is not conclusive.

If the check fails probably we want to return whatever the inner mapper returned.

Reply:
You are right, for some reason I thought that not including the link if the check fails for any reason was OK to skip the changelog link... It might be OK but doing it silently can be very frustrating. Thanks for pointing it out.

Concerning the value to return, I am not sure if I prefer including a link which may be broken (the value that the inner mapper returned) or no link (an empty string). WDYT?

Reply:
Whatever we decide to do when links cannot be checked due to unexpected errors, that situation should be handled by the mapper and not by the check function, so I changed the function signature as you suggested:[6f5cbde](https://github.com/newrelic/release-toolkit/pull/134/commits/6f5cbde1ffe7074c0185bed72543d9d8264ec1e3)

Regarding the potentially broken links, there is a risk of generating those when the check cannot be performed due to network issues, but as long as we properly warn the users it could be a good a approach.

Reply:
I think there is no good solution for this one. If we remove the link (return empty) on error, we could be reflecting transient errors on a permanent state (the changelog). If we error out and abort, we would be making the tool unreliable. If we do nothing (return link as-is from the inner mapper), we could be creating an inconsistency.

I don't like either, but I think returning as-is and logging an error is the one I dislike less.